### PR TITLE
Add configuration using configor.

### LINKED ISSUE #39: Create a better configuration system based on yaml files
We can try do golang libraries to do this once we agree on which one to use.

### PR DESCRIPTION
**Pull request by stephenwithav**
This PR replaces #78.

Adds support for command-line flags, YAML configuration, and configuration via environment variables.

Closes #39.

### CODE REVIEW

**Comment by vimukthi-git** [comment]
@stephenwithav Looks good. Please add the dependancy to Godeps.

**Comment by stephenwithav** [comment]
I didn't add everything in `vendor`--just the `Goeps.json` file.

File: Godeps/Godeps.json [hunk 1]
```diff
@@ -1,17 +1,54 @@
...
 			"ImportPath": "github.com/satori/go.uuid",
 			"Comment": "v1.1.0-8-g5bf94b6",
 			"Rev": "5bf94b69c6b68ee1b541973bb8e1144db23a194b"
+		},
+		{
+			"ImportPath": "github.com/stretchr/objx",
+			"Rev": "1a9d0bb9f541897e62256577b352fdbc1fb4fd94"
+		},
+		{
+			"ImportPath": "github.com/stretchr/testify/assert",
```
Comment (vimukthi-git) [inline Godeps/Godeps.json]: Why are these testify things getting added to the file?

File: Godeps/Godeps.json [hunk 2]
```diff
@@ -1,17 +1,54 @@
...
+			"Comment": "v1.1.4-66-gf6abca5",
+			"Rev": "f6abca593680b2315d2075e0f5e2a9751e3f431a"
+		},
+		{
+			"ImportPath": "github.com/stretchr/testify/mock",
```
Comment (vimukthi-git) [inline Godeps/Godeps.json]: Here also

**Comment by vimukthi-git** [comment]
@stephenwithav you can commit the vendor directory its ok. But could you find a way to avoid adding test dependancies?

File: Godeps/Godeps.json (hunk 1 above)
Comment (vimukthi-git) [inline Godeps/Godeps.json]: Just realised this is my fault. So I will correct.

File: Godeps/Godeps.json (hunk 2 above)
Comment (vimukthi-git) [inline Godeps/Godeps.json]: Just realised this is my fault. So I will correct.

**Comment by stephenwithav** [comment]
Rebased, vendored packages included, and ready for merge.

**Pull request merged**